src/firmware/shared/physics.h

```diff
@@ -103,3 +103,18 @@ void rotate(float speed3[2], float angle);
 float dot_product(const float vec1[], const float vec2[], const int size);
 
 float dot2D(float vec1[2], float vec2[2]);
+
+/**
+ * Function calculates the final speed at the end of a displacement given an initial speed
+ * and constant acceleration
+ *
+ * @pre All values are positive
+ *
+ * @param initial_speed [in] The initial speed in meters/second
+ * @param displacement [in] The total displacement accelerated over in meters
+ * @param acceleration [in] The constant acceleration over the displacement in
+ * meters/second^2
+ * @return The final speed in meters/second
+ */
+float shared_physics_calculateFinalSpeedFromDisplacementInitialSpeedAndAcceleration(
```

Comment:
nit: This name is very descriptive but also very redundant with the parameter information (it completely describes all the params). I don't think that's necessary in this case, so I think `shared_physics_calculateFinalSpeed` would be sufficient?

@garethellis0 @jonathanlew @akhilveeraghanta for additional opinions

Reply:
Renamed 👍